test/ui/viewJobRoleTests.ts

```diff
@@ -0,0 +1,98 @@
+import { Builder, WebDriver } from 'selenium-webdriver';
+import webdriver from 'selenium-webdriver';
+import { expect } from 'chai';
+import * as chrome from 'selenium-webdriver/chrome';
+
+describe('Job Roles Test', function () {
+
+    let driver: WebDriver;
+
+    before(async function () {
+        driver = new Builder()
+            .forBrowser('chrome')
+            .setChromeOptions(new chrome.Options())
+            .build();
+
+    });
+
+    after(async function () {
+        try {
+            await driver.quit();
+        } catch (error) {
+            console.error('Error quitting the driver:', error);
+        }
+    });
+
+    it('Page should load and title text should display', async function () {
+        const url: string = process.env.UI_TEST_URL || 'https://2qr8mnb3c3.eu-west-1.awsapprunner.com/openJobRoles';
+        await driver.get(url);
+        const title = await driver.findElement(webdriver.By.css('h1')).getText();
+
+        expect(title).to.equal('Open Job Roles');
+    });
+
+    it('Job role table should populate', async function () {
+        const url: string = process.env.UI_TEST_URL || 'https://2qr8mnb3c3.eu-west-1.awsapprunner.com/openJobRoles';
+        await driver.get(url);
+        const data = await driver.findElement(webdriver.By.css('td')).getText();
+
+        expect(data).to.not.equal(null);
+    });
+
+    it('Location data should only show valid locations', async function () {
+        const url: string = process.env.UI_TEST_URL || 'https://2qr8mnb3c3.eu-west-1.awsapprunner.com/openJobRoles';
+        await driver.get(url);
+
+        const names: string[] = ["BELFAST", "LONDON", "TORONTO", "BIRMINGHAM", "GDANSK"];
+
+        const table = await driver.findElement(webdriver.By.css('table'));
```

Comment:
Ah, unfortunately, this isn't reliable either. Anything that relies on page structure could break with further requirements. Please look into adding the ID's like we chatted about and using them as your selectors. It should be less work than all this code :D